plugins/language_java.lua

```diff
@@ -6,19 +6,21 @@ syntax.add {
   files = { "%.java$" },
   comment = "//",
   patterns = {
-    { pattern = "//.-\n",               type = "comment"  },
-    { pattern = { "/%*", "%*/" },       type = "comment"  },
-    { pattern = { '"', '"', '\\' },     type = "string"   },
-    { pattern = { "'", "'", '\\' },     type = "string"   },
-    { pattern = "'\\x%x?%x?%x?%x'",         type = "string"   }, -- character hexadecimal escape sequence
-    { pattern = "'\\u%x%x%x%x'",            type = "string"   }, -- character unicode escape sequence
-    { pattern = "'\\?.'",                   type = "string"   }, -- character literal
-    { pattern = "-?0x%x+",              type = "number"   },
-    { pattern = "-?%d+[%d%.eE]*f?",     type = "number"   },
-    { pattern = "-?%.?%d+f?",           type = "number"   },
-    { pattern = "[%+%-=/%*%^%%<>!~|&]", type = "operator" },
-    { pattern = "[%a_][%w_]*%f[(]",     type = "function" },
-    { pattern = "[%a_][%w_]*",          type = "symbol"   },
+    { pattern = "//.*",                                         type = "comment"  },
+    { pattern = { "/%*", "%*/" },                               type = "comment"  },
+    { pattern = { '"', '"', '\\' },                             type = "string"   },
+    { pattern = { "'", "'", '\\' },                             type = "string"   },
+    { pattern = "'\\x%x?%x?%x?%x'",                             type = "string"   }, -- character hexadecimal escape sequence
+    { pattern = "'\\u%x%x%x%x'",                                type = "string"   }, -- character unicode escape sequence
+    { pattern = "'\\?.'",                                       type = "string"   }, -- character literal
+    { pattern = "-?0x%x+",                                      type = "number"   },
+    { pattern = "-?%d+[%d%.eE]*f?",                             type = "number"   },
+    { pattern = "-?%.?%d+f?",                                   type = "number"   },
+    { pattern = "[%+%-=/%*%^%%<>!~|&]",                         type = "operator" },
+    { pattern = "[%a_][%w_]*%f[(]",                             type = "function" },
+    { regex   = "(?:\\w+\\.?)+(?=\\s+\\w++\\s+\\=\\s+)",        type = "function" }, -- Class name in class instance
```

Comment:
What is this supposed to match?

Reply:
It's something that Eclipse does, it highlights the name of the class when creating an object.

Reply:
![image](https://github.com/user-attachments/assets/8f06e2be-52d6-4d26-96da-ed35386e5fac)

Reply:
Then I guess you should make the spaces around `=` optional.